### ISSUE TITLE
Auto-Initializing Deep Neural Networks with GradInit

### ISSUE THREAD

**Issue by kayuksel** [pytorch/pytorch#52626]
## 🚀 Feature

Auto-Initializing Deep Neural Networks with GradInit for Stable and Efficient Training

GradInit is an automated and architecture agnostic method for initializing deep neural networks based on a simple heuristic: 
the varianc e of each network layer is adjusted so that a single step of optimizer results in the smallest possible loss value.

## Motivation

Hypertuning the optimal weight initialization for a deep neural network is often a nightmare.

## Pitch

I don't want to worry about setting the optimal initialization for each module of my network.

## Additional context

Code: https://github.com/zhuchen03/gradinit
Paper: https://arxiv.org/abs/2102.08098

cc @albanD @mruberry @jbschlosser

**Comment by zhuchen03** [comment]
Thanks for opening the feature request! It may be achievable without adding any new features to PyTorch. I will try to get a version soon.

**Comment by albanD** [comment]
Hi,

The general policy for new features can be found in this other comment https://github.com/pytorch/pytorch/issues/48520#issuecomment-736602300
Given that the paper was just released, it is unlikely to meet these criteria though.
We are happy to help provide any function that prevents you from easily implementing it as a third party library.

Also for the question in the other repo, it should be fairly easy to provide that third party implementation as a package on pypi so that you can simply pip install it.

**Comment by zhuchen03** [comment]
Got it, thanks for offering the help! Let me first get a handy package out. Currently I think PyTorch has provided enough functions to make the implementation (neural) architecture-agnostic. Let's reconsider the feature request later.

**Comment by albanD** [comment]
Sounds good.
Closing for now then. Feel free to ask any question here or on the forum and re-open the issue when we should reconsider!
